app/services/decidim/sms_gateway_service.rb

```diff
@@ -8,28 +8,46 @@ class SmsGatewayService
     attr_reader :mobile_phone_number, :code
 
     def initialize(mobile_phone_number, code, sms_gateway_context = {})
-      Rails.logger.debug { "#{mobile_phone_number} - #{code}" }
-
       @mobile_phone_number = mobile_phone_number
       @code = code
       @organization_name = sms_gateway_context[:organization]&.name
       @url = fetch_configuration(:url)
-      @username = fetch_configuration(:username)
-      @password = fetch_configuration(:password)
+      @username = fetch_configuration(:username, required: false)
+      @password = fetch_configuration(:password, required: false)
+      @mb_account_id = fetch_configuration(:mb_account_id, required: false)
+      @mb_api_key = fetch_configuration(:mb_api_key, required: false)
       @message = sms_message
       @type = "sms"
     end
 
     def deliver_code
-      url = URI("#{@url}?u=#{@username}&p=#{@password}&t=#{@message}&n=#{@mobile_phone_number}&f=#{@type}")
+      url, request = build_request
       https = Net::HTTP.new(url.host, url.port)
       https.use_ssl = true
-      request = Net::HTTP::Get.new(url)
       https.request(request)
-
       true
     end
 
+    def build_request
+      if @url.include?("services.message-business.com")
```

Comment:
Prefer to define a dedicated class implementing an interface SmsGateway with method `request` callable rather than if / else condition